modules/datasets/src/track_alov.cpp

```diff
@@ -0,0 +1,384 @@
+		{
+			string out;
+			char videoNum[9];
+			sprintf(videoNum, "%u", videoID+1);
+			out = rootPath + "/alov300++_rectangleAnnotation_full/" + sectionNames[sectionID] + "/" + sectionNames[sectionID] + "_video";
+
+			for (unsigned int i = 0; i < 5 - strlen(videoNum); ++i)
+			{
+				out += "0";
+			}
+
+			out += videoNum;
+			out += ".ann";
+			return out;
+		}
+
+        inline bool fileExists(const std::string& name)
+        {
+            struct stat buffer;
+            return (stat(name.c_str(), &buffer) == 0);
+        }
+
+		void TRACK_alovImpl::loadDataset(const string &rootPath)
+        {
+            vector <int> datasetsLengths;
+
+			printf("ALOV300++ Dataset Initialization...\n");
+
+			//Load frames
+			//Loop for all sections of ALOV300++ (14 sections)
+			for (int i = 0; i < 14; i++)
+			{
+				//Loop for all videos in section
+				for (int k = 0; k < sectionSizes[i]; k++)
+				{
+					vector <Ptr<TRACK_alovObj> > objects;
+
+					//Make a list of datasets lengths
+					int currFrameID = 0;
+
+					for (;;)
+					{
+						currFrameID++;
+						string fullPath = fullFramePath(rootPath, i, k, currFrameID);
+						if (!fileExists(fullPath))
+							break;
+
+						//Make ALOV300++ Object
+						Ptr<TRACK_alovObj> currObj(new TRACK_alovObj);
+						currObj->imagePath = fullPath;
+						currObj->id = currFrameID;
+
+						currObj->gtbb.push_back(Point2d(0, 0));
+						currObj->gtbb.push_back(Point2d(0, 0));
+						currObj->gtbb.push_back(Point2d(0, 0));
+						currObj->gtbb.push_back(Point2d(0, 0));
+
+						//Add object to storage
+						objects.push_back(currObj);
+
+					}
+
+					datasetsLengths.push_back(currFrameID - 1);
+					data.push_back(objects);
+				}
+			}
+
+			//Load annotations
+			//Loop for all sections of ALOV300++ (14 sections)
+			int currDatasetID = 0;
+			for (int i = 0; i < 14; i++)
+			{
+				//Loop for all videos in section
+				for (int k = 0; k < sectionSizes[i]; k++)
+				{
+					currDatasetID++;
+
+					//Open dataset's ground truth (annotation) file
+					string annoPath = fullAnnoPath(rootPath, i, k);
+					ifstream annoList(annoPath.c_str());
+					if (!annoList.is_open())
+					{
+						printf("Error: Can't open annotation file *.ANN!!!\n");
+						break;
+					}
+
+					//Ground Truth data
+					int n = 0;
+					double	x1 = 0, y1 = 0,
+						x2 = 0, y2 = 0,
+						x3 = 0, y3 = 0,
+						x4 = 0, y4 = 0;
+
+					do
+					{
+						//Make ALOV300++ Object
+						string tmp;
+
+						getline(annoList, tmp);
+						std::istringstream in(tmp);
+						in >> n >> x1 >> y1 >> x2 >> y2 >> x3 >> y3 >> x4 >> y4;
+
+						Ptr<TRACK_alovObj> currObj = data[currDatasetID-1][n-1];
+
+						currObj->gtbb.clear();
+						currObj->gtbb.push_back(Point2d(x1, y1));
+						currObj->gtbb.push_back(Point2d(x2, y2));
+						currObj->gtbb.push_back(Point2d(x3, y3));
+						currObj->gtbb.push_back(Point2d(x4, y4));
+
+					} while (!annoList.eof());
```

Comment:
It is better to use `annoList.good()` instead of `!annoList.eof()` since the former check also the status of `failbit` and `badbit`